src/decode/mod.rs

```diff
@@ -77,6 +84,12 @@ impl TraceDecoderBuilder {
         self.kind.match_platform()?;
         match self.kind {
             TraceDecoderKind::LibIPT => Ok(Box::new(LibIPTTraceDecoder::new())),
+            TraceDecoderKind::YkPT => {
+                #[cfg(decoder_ykpt)]
+                return Ok(Box::new(YkPTTraceDecoder::new()));
```

Comment:
`return` not needed

Reply:
But it is. It's a limitation/bug in Rust's parser.

```
   Compiling hwtracer v0.1.0 (/home/vext01/research/hwtracer)
error: expected `;`, found `#`
  --> src/decode/mod.rs:94:54
   |
94 |                 Ok(Box::new(YkPTTraceDecoder::new()))
   |                                                      ^ help: add `;` here
95 |                 #[cfg(not(decoder_ykpt))]
   |                 - unexpected token
```

Reply:
Egads!